src/passes/pass.cpp

```diff
@@ -546,27 +546,33 @@ static void dumpWast(Name name, Module* wasm) {
 
 void PassRunner::run() {
   static const int passDebug = getPassDebug();
-  if (!isNested && (options.debug || passDebug)) {
+  // Emit logging information when asked for. At passDebug level 1+ we log
+  // the main passes, while in 2 we also log nested ones. Note that for
+  // nested ones we can only emit their name - we can't validate, or save the
+  // file, or print, as the wasm may be in an intermediate state that is not
+  // valid.
+  if (options.debug || (passDebug == 2 || (passDebug && !isNested))) {
```

Comment:
What does `options.debug` do here?

Reply:
It's a way for a specific PassRunner object itself to be marked as debug (instead of globally using the env var). Not sure if it's useful actually.